### ISSUE TITLE
eyaml backend not decrypting folded blocks in yaml

### ISSUE THREAD

**Issue by znorris** [TomPoulton/hiera-eyaml#136]
eyaml backend not decrypting data when in folded blocks like the example given:
```
password: >
            ENC[PKCS7,Y22exl+OvjDe+drmik2XEeD3VQtl1uZJXFFF2NnrMXDWx0csyqLB/2NOWefv
            NBTZfOlPvMlAesyr4bUY4I5XeVbVk38XKxeriH69EFAD4CahIZlC8lkE/uDh
            jJGQfh052eonkungHIcuGKY/5sEbbZl/qufjAtp/ufor15VBJtsXt17tXP4y
            l5ZP119Fwq8xiREGOL0lVvFYJz2hZc1ppPCNG5lwuLnTekXN/OazNYpf4CMd
            /HjZFXwcXRtTlzewJLc+/gox2IfByQRhsI/AgogRfYQKocZgFb/DOZoXR7wm
            IZGeunzwhqfmEtGiqpvJJQ5wVRdzJVpTnANBA5qxeA==]
```

The backend will decrypt them if kept as a single string on one line.

**Comment by elyscape** [comment]
What version are you using?  That functionality works on my machine.

**Comment by znorris** [comment]
Version 2.0.3. I saw it was the latest release.
I don't get any specific errors. It outputs the file but the value is still encrypted.

**Comment by elyscape** [comment]
Does it decrypt properly from the command line? `eyaml decrypt -f filename`
If it does not, try adding the `--verbose` flag and see if it prints anything that might be useful.

**Comment by gtmtech** [comment]
Agreed - the actual format of yaml is handled by the underlying YAML parser and is nothing to do with hiera or hiera-eyaml. I would hazard a guess that something else is afoot - does your file have a .yaml or .eyaml extension - can you give more info? running --debug and --verbose as above may help - --debug in particular activates the debug output within hiera-eyaml

**Comment by znorris** [comment]
Here is the (sanitized) output of 
```
eyaml decrypt -f /usr/local/src/project/provision/puppet/hiera/common.yaml --trace --verbose
```

```
[hiera-eyaml-core] Dump of eyaml tool options dict:
[hiera-eyaml-core] --------------------------------
[hiera-eyaml-core]           (Symbol) encrypt_method     =           (String) pkcs7
[hiera-eyaml-core]           (Symbol) version            =       (FalseClass) false
[hiera-eyaml-core]           (Symbol) verbose            =        (TrueClass) true
[hiera-eyaml-core]           (Symbol) trace              =        (TrueClass) true
[hiera-eyaml-core]           (Symbol) quiet              =       (FalseClass) false
[hiera-eyaml-core]           (Symbol) help               =       (FalseClass) false
[hiera-eyaml-core]           (Symbol) string             =         (NilClass)
[hiera-eyaml-core]           (Symbol) file               =           (String) /usr/local/src/project/provision/puppet/hiera/common.yaml
[hiera-eyaml-core]           (Symbol) eyaml              =         (NilClass)
[hiera-eyaml-core]           (Symbol) stdin              =       (FalseClass) false
[hiera-eyaml-core]           (Symbol) pkcs7_private_key  =           (String) ./keys/private_key.pkcs7.pem
[hiera-eyaml-core]           (Symbol) pkcs7_public_key   =           (String) ./keys/public_key.pkcs7.pem
[hiera-eyaml-core]           (Symbol) pkcs7_subject      =           (String) /
[hiera-eyaml-core]           (Symbol) file_given         =        (TrueClass) true
[hiera-eyaml-core]           (Symbol) trace_given        =        (TrueClass) true
[hiera-eyaml-core]           (Symbol) verbose_given      =        (TrueClass) true
[hiera-eyaml-core]           (Symbol) executor           =            (Class) Hiera::Backend::Eyaml::Subcommands::Decrypt
[hiera-eyaml-core]           (Symbol) source             =           (Symbol) file
[hiera-eyaml-core]           (Symbol) input_data         =           (String) apache::default_vhost:     true

                   mysql::server::package_name:  'mysql-server-5.6'
                   mysql::server::root_password: ENC[PKCS7,MIIBiQYJKoZIhvcNAQcDoIIBejCCAXYCAQAxggEhMIIBHQIBADAFMAACAQEwDQYJKoZIhvcNAQEBBQAEggEANEFi5Ql72kADp7mUd1lVgO8Zl9Lgt/qxDBKhiDoOBXZvUMXXXXKnh42xasB5h67W5SiSWcWJVH/leY0mB/XW3NsbzQo9iffuyn/cyln+CCfDLz1FAjrnesXk8MUebqxXOR1gIaCge6CpTe/oFcz6nVYLY0DCUQEWqFHg6yWBv/Ri6s8lAiwtLGjoltG5egATcP/ul7BMsr9nJBoqvth+xUCjfXyrdoaqTOLjRC0bGASUfKp7T7FLwtRrW1UScWMtyEj4wPFuKnjdT+oCmqgEdG050ECQmtC0L2vjznhFSDiXuncIWixaq21nzJn6k+ooRlalJfgzYqRAjJIF+rvoczBMBgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBCqx33cbpDMC6PGIhRKk+6LgCARFcJ/rC+PXcvH87jmfjSE+kRKQ4vaNR18He579ytd9A==]
                   mysql::server::root_password_block: >
                       ENC[PKCS7,MIIBiQYJKoZIhvcNAQcDoIIBejCCAXYCAQAxggEhMIIBHQIBADAFMAACAQEw
                       DQYJKoZIhvcNAQEBBQAEggEAAHPNgcuZrqhuMD6Wh0tP6jBqhKe7Q2Lr/7bM
                       RBxdrEksepX2pWKc7/cFR7OEQBSw8/4TyhsZiPBhHzDxHcgd4At6mf1wal2i
                       /w6v8vcTC4tBlGAYT4Sur8nt2Zp9IVk1CqOwt1UhMSGmdWhgvn2ZFcGewl+F
                       Snb7D7GPh2bWuqMEzHbUSnRDMWWKmghgIMl3agJz8vcdaMyjI1tN3H/99wRO
                       OBXZA1fV0bZFgtZ87AebQrjRrHBxd4+5m4iYSXV5e6wjhPoTkOuCn1JWX2HV
                       1SRnJrjgHX+pU9yWAX3cDGOwxAY21F9P45hrIDt+2enghSlemNHApwCppEQT
                       TOlG/zBMBgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBCAfDnaMkFsi7hqgBFq
                       8R5HgCD54rnMbvD6tu9MwUojr6Wk+C671LU1hu8B8bm37VodFg==]

[hiera-eyaml-core] --------------------------------
apache::default_vhost:     true
apache::default_ssl_vhost: false
apache::manage_user:       false
apache::manage_group:      false
apache::mpm_module:        prefork
apache::user:              "%{hiera('project::user')}"
apache::group:             "%{hiera('project::group')}"


mysql::server::package_name:  'mysql-server-5.6'
mysql::server::root_password: THEPASSWORDWASDECRYPTEDHERE
mysql::server::root_password_block: >
    ENC[PKCS7,MIIBiQYJKoZIhvcNAQcDoIIBejCCAXYCAQAxggEhMIIBHQIBADAFMAACAQEw
    DQYJKoZIhvcNAQEBBQAEggEAAHPNgcuZrqhuMD6Wh0tP6jBqhKe7Q2Lr/7bM
    RBxdrEksepX2pWKc7/cFR7OEQBSw8/4TyhsZiPBhHzDxHcgd4At6mf1wal2i
    /w6v8vcTC4tBlGAYT4Sur8nt2Zp9IVk1CqOwt1UhMSGmdWhgvn2ZFcGewl+F
    Snb7D7GPh2bWuqMEzHbUSnRDMWWKmghgIMl3agJz8vcdaMyjI1tN3H/99wRO
    OBXZA1fV0bZFgtZ87AebQrjRrHBxd4+5m4iYSXV5e6wjhPoTkOuCn1JWX2HV
    1SRnJrjgHX+pU9yWAX3cDGOwxAY21F9P45hrIDt+2enghSlemNHApwCppEQT
    TOlG/zBMBgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBCAfDnaMkFsi7hqgBFq
    8R5HgCD54rnMbvD6tu9MwUojr6Wk+C671LU1hu8B8bm37VodFg==]
```

**Comment by elyscape** [comment]
Are there any tab characters in the file?  Run `grep \t /usr/local/src/project/provision/puppet/hiera/common.yaml -c` and see what it prints.  eyaml doesn't seem to like leading tab characters.

**Comment by znorris** [comment]
No tab characters were found.

**Comment by elyscape** [comment]
Well that's really weird because tab characters definitely break it the way you describe.  Are there any carriage returns in your file? `grep \r /usr/local/src/project/provision/puppet/hiera/common.yaml -c`.

**Comment by elyscape** [comment]
Sorry, use this command: `grep \r /usr/local/src/project/provision/puppet/hiera/common.yaml --binary -c`

**Comment by znorris** [comment]
Yes, 190 of them. Note, I have to modify your command a bit ```grep $'\r' /file```

**Comment by elyscape** [comment]
Okay, cool.  I have a fix for that.  Submitting a PR now.

**Comment by elyscape** [comment]
A temporary workaround while we wait for the PR to hit would of course be to remove all carriage returns from your file.

**Comment by znorris** [comment]
Thank you. I appreciate it.

**Comment by gtmtech** [comment]
Good spot @elyscape

**Comment by beddari** [comment]
Works fine :+1:

**Comment by TomPoulton** [comment]
Fixed in v2.0.7, thanks @elyscape, @znorris, @gtmtech
